internal/ingress/controller/controller.go

```diff
@@ -995,7 +997,15 @@ func (n *NGINXController) createUpstreams(data []*ingress.Ingress, du *ingress.B
 					endp, err := n.serviceEndpoints(svcKey, port.String())
 					if err != nil {
 						klog.Warningf("Error obtaining Endpoints for Service %q: %v", svcKey, err)
+						n.metricCollector.IncOrphanIngress(ing.Namespace, ing.Name, orphanMetricLabelNoService)
 						continue
+					} else {
+						n.metricCollector.DecOrphanIngress(ing.Namespace, ing.Name, orphanMetricLabelNoService)
+					}
+					if len(endp) == 0 {
```

Comment:
This thing I don't like (double check `len(endp)`), but I don't found the way to not passing to`n.serviceEndpoints` additional arguments and check metric in it (maybe, it is not a problem and all seems ok)

Reply:
hum, I think serviceEndpoints being a method of "n" you can call n.metricCollector inside it. What would be the extra arguments needed there?

Reply:
Extra argument would be `ingress.Ingress` (for two first arguments of `n.metricCollector.DecOrphanIngress`) to get ingress name and namespace